### ISSUE TITLE
About target_category

### ISSUE THREAD

**Issue by m250317460** [jacobgil/pytorch-grad-cam#110]
Would you mind explaining the function of this parameter in detail？

**Comment by chaomartin** [comment]
yeah, i also have the problem about how to specific a category?   Can you give us some guidance?

**Comment by chaomartin** [comment]
I think the default category index is in the range of 1-999, corresponding to the category labels in the ImageNet dataset. but how to set the category index if i want to use CAM in COCO dataset? I have the related document and the categories of Imagenet datasets do not completely include the categories of COCO datasets. Therefore, how to solve this problem?

**Comment by jacobgil** [comment]
This code assumes a classification setting, where the model has N output scores for N categories.
For imagenet indeed N=1000.
For other datasets it will be different.
If you're not sure how many outputs your model has - you can print it. I think COCO has 80 categories, so models trained on it should have 80 outputs.

The CAM visualization is computed for one of these categories - you control which one.
If you specify category 10 for example, it will compute the visualization for the 10th output, which might look different than the visualization of the other categories.

If you set category_index = None, it will use the highest scoring category (which of course won't always be correct) and compute the visualization for that.

**Comment by chaomartin** [comment]
First of all, I appreciate your response.  And I agree with what you talk about. Right now, i plan to design a classification network with the customized categories output channels and get the related weights.  I can get the matched CAM with the fine-tuned weights.

在2021-07-09 20:02:36，Jacob ***@***.***写道：

This code assumes a classification setting, where the model has N output scores for N categories.
For imagenet indeed N=1000.
For other datasets it will be different.
If you're not sure how many outputs your model has - you can print it. I think COCO has 80 categories, so models trained on it should have 80 outputs.

The CAM visualization is computed for one of these categories - you control which one.
If you specify category 10 for example, it will compute the visualization for the 10th output, which might look different than the visualization of the other categories.

If you set category_index = None, it will use the highest scoring category (which of course won't always be correct) and compute the visualization for that.
